ts/Core/Axis/Axis.ts

```diff
@@ -2183,7 +2183,12 @@ class Axis {
 
         fireEvent(this, 'trimTicks');
 
-        if (!this.isLinked) {
+        if (
+            !this.isLinked ||
+            // Linked non-grid axes should trim ticks, #21743.
+            // Grid axis has custom handling of ticks.
+            !this.grid
+        ) {
```

Comment:
Now do we need the `isLinked` test at all? Wouldn't it be enough with `if (!this.grid)` ?

Reply:
Could be, I can test more 👍